File: cft-integration/src/de/proskor/fel/impl/EventTypeImpl.java
package de.proskor.fel.impl;

import java.util.ArrayList;
import java.util.List;

import de.proskor.fel.container.EventTypeContainer;
import de.proskor.fel.event.EventInstance;
import de.proskor.fel.event.EventType;


public class EventTypeImpl implements EventType {
	private String name;
	private String guid;
	private int id;
	private String author;
	private String description;

	private List<EventInstance> instances;
	private EventTypeContainer container;

	@Override
	public String getName() {
		return name;
	}

	
	@Override
	public String getGuid() {
		return guid;
	}
	
	@Override
	public int getId() {
		return id;
	}

	@Override
	public String getAuthor() {
		return author;
	}

	@Override
	public String getDescription() {
		return description;
	}

	@Override
	public List<EventInstance> getInstances() {
		return instances;
	}

	@Override
	public void addInstance(EventInstance eventInstance) {
		if (!instances.contains(eventInstance))
			instances.add(eventInstance);
	}

	@Override
	public EventTypeContainer getContainer() {
		return container;
	}

	@Override
	public String toString() {
		return "Event: "+name+"; Container: "+container+";  ID: "+id+";  guid: "+guid + "; author: " + author + "; description: " + description;
	}
	
	public EventTypeImpl(String name, EventTypeContainer container, String author, String description, String guid, int id) {
		this.name = name;
		this.container = container;		
		this.author = author;
		this.description = description;
		this.guid = guid;
		this.id = id;
		instances = new ArrayList<EventInstance>();
		
		container.addEvent(this);
	}
}

File: ea-fel/src/de/proskor/fel/impl/ea/EventTypeImpl.scala
package de.proskor.fel.impl.ea

import java.util.{List => JavaList}
import collection.JavaConversions._
import de.proskor.fel.EventType
import de.proskor.automation.Element
import de.proskor.fel.EventInstance
import de.proskor.automation.Repository

class EventTypeImpl(peer: Element) extends EventType {
  override def getId: Int = peer.id
  override def getGuid: String = "TODO"
  override def getName: String = peer.name
  override def getAuthor: String = "TODO"
  override def getDescription: String = "TODO"

  override def getInstances: JavaList[EventInstance] = (for {
      connector <- peer.connectors
      if connector.target == this && connector.stereotype == "instanceOf"
      instance = connector.source
    } yield new EventInstanceImpl(instance)).toList
  
  override def addInstance(instance: EventInstance) {
  //  val models = Repository.instance.models
  //  val fel = models.find(_.name == "FEL") getOrElse models.add("FEL", "Package")
    val connector = peer.connectors.add("instanceOf", "instanceOf")
    connector.source = instance.asInstanceOf[EventInstanceImpl].peer
    connector.target = this.peer
  }
}